components/camel-mail/src/main/java/org/apache/camel/component/mail/MailBinding.java

```diff
@@ -368,13 +368,18 @@ protected void extractAttachmentsFromMultipart(Multipart mp, Map<String, Attachm
                         }
                         map.put(fileName, camelAttachment);
                     } else {
-                        LOG.warn("Cannot extract duplicate file attachment: {}.", fileName);
+                        handleDuplicateAttachmentFilename(fileName);
```

Comment:
I would avoid creating a method for one line exception throw

Reply:
The idea was, that custom bindings could override the error handling, i.e. only log an error.